test/support/page_objects/header_page.js

```diff
@@ -144,6 +144,23 @@ export default class HeaderPage {
     .click();
   }
 
+  async waitUntilLoadingHasFinished() {
+    try {
+      await this.isGlobalLoadingIndicatorVisible();
```

Comment:
I think this `await this.isGlobalLoadingIndicatorVisible();` call is going to take 10 second (defaultFindTimeout) to timeout if the loading indicator isn't there (like in the case where it was really fast and already became hidden before this check).  Slow is better than flaky, but I think findTestSubject should allow passing a timeout and set it to something like 1 or 2 seconds for this call.